File: archive-common/src/main/java/site/archive/common/cache/CacheType.java
package site.archive.common.cache;

import lombok.Getter;

@Getter
public enum CacheType {

    BANNERS(CacheInfo.BANNERS, CacheInfo.BANNERS_GET, 1),
    ARCHIVE_COMMUNITIES(CacheInfo.ARCHIVE_COMMUNITIES, CacheInfo.ARCHIVE_COMMUNITIES_GET, 200)
    ;

    private final String name;
    private final int expireAfterWrite; // SECONDS TimeUnit
    private final int maximumSize;

    CacheType(String name, int expireAfterWrite, int maximumSize) {
        this.name = name;
        this.expireAfterWrite = expireAfterWrite;
        this.maximumSize = maximumSize;
    }

    public static class CacheInfo {

        public static final String BANNERS = "banners";
        static final int BANNERS_GET = 3_600;   // 1 hour

        public static final String ARCHIVE_COMMUNITIES = "archive-communities";
        static final int ARCHIVE_COMMUNITIES_GET = 30;

        private CacheInfo() {
        }

    }

}

File: archive-common/src/main/kotlin/site/archive/common/cache/CacheType.kt
package site.archive.common.cache

enum class CacheType(val key: String, val expireAfterWrite: Int, val maximumSize: Int) {

    BANNERS(CacheInfo.BANNERS, CacheInfo.BANNERS_GET_EXPIRE_TIME, CacheInfo.BANNERS_CACHE_SIZE),
    ARCHIVE_COMMUNITIES(CacheInfo.ARCHIVE_COMMUNITIES, CacheInfo.ARCHIVE_COMMUNITIES_GET_EXPIRE_TIME, CacheInfo.ARCHIVE_COMMUNITIES_CACHE_SIZE)
    ;

    class CacheInfo {
        companion object {
            const val BANNERS: String = "banners"
            const val BANNERS_GET_EXPIRE_TIME: Int = 3_600
            const val BANNERS_CACHE_SIZE: Int = 1

            const val ARCHIVE_COMMUNITIES: String = "archive-communities"
            const val ARCHIVE_COMMUNITIES_GET_EXPIRE_TIME: Int = 30
            const val ARCHIVE_COMMUNITIES_CACHE_SIZE: Int = 200
        }
    }

}